tensorflow/compiler/xla/service/mlir_gpu/BUILD

```diff
@@ -39,9 +39,9 @@ cc_library(
     deps = [
         ":failover_compiler",
         "//tensorflow/compiler/xla/service:compiler",
-        "//tensorflow/compiler/xla/service/gpu:gpu_compiler_impl",
         "//tensorflow/compiler/xla/service/gpu:gpu_constants",
         "//tensorflow/compiler/xla/service/gpu:ir_emission_utils",
+        "//tensorflow/compiler/xla/service/gpu:nvptx_compiler",
```

Comment:
I don't think this will work, because the nvptx_compiler target registers `stream_executor::cuda::kCudaPlatformId` (nvptx_compiler_registration.cc).
mlir_compiler.cc will try to do the same and fail. Please keep the split between nvptx_compiler and nvptx_compiler_impl.

Reply:
will revise.